### ISSUE TITLE
Beta-testing 'mini.hues'

### ISSUE THREAD

**Issue by echasnovski** [echasnovski/mini.nvim#319]
Please leave your feedback about new [mini.hues](https://github.com/echasnovski/mini.nvim/blob/main/readmes/mini-hues.md) module here. Feel free to either add new comment or positively upvote existing one.

Some things I am interested to find out (obviously, besides bugs):

- Do all supported integrations have proper highlight groups?
- Are configuration and function names intuitive enough?
- Are default values of settings convenient for you?
- Is documentation and examples clear enough?

Thanks!

**Comment by marcusandre** [comment]
@echasnovski I have been looking forward to this for a quite some time. Many colour schemes are too colourful for me (when it comes to highlighting). I like it a bit more stripped down, so I will try to recreate my favourite ones. The `n_hues` option is naturally a big win for me. Thanks for creating another useful plugin! 🙂

**Comment by echasnovski** [comment]
Thanks for the feedback!

> The `n_hues` option is naturally a big win for me. Thanks for creating another useful plugin! slightly_smiling_face

Beware that it will only replace non-base colors and not remove them. So it it will be less like a rainbow but still colored (like in a video demo). For anything more involved I'd recommend checking 'mini.colors' (like filtering out colors within certain hue ranges).

This is mostly because I also prefer using more colors for more information. But, of course, not too much. I found that 8 hues were just enough (original attempt with 6 did not allow describe all needed information).

**Comment by marcusandre** [comment]
Yes, that's a nice hint. I will play with that for a while. Currently trying to come up with something that looks like _nord_, but suits my eyes even better.

<img width="1150" alt="Bildschirmfoto 2023-05-07 um 18 58 58" src="https://user-images.githubusercontent.com/102828/236691962-e1b9b4a2-ed89-4fb4-8cb6-1854d4b04abb.png">
<img width="1140" alt="Bildschirmfoto 2023-05-07 um 19 04 09" src="https://user-images.githubusercontent.com/102828/236691969-bfeb84dc-6dce-4ab3-9048-02c6979ef2a0.png">

---

Two comments I already have: 
1. Is there a simple way to construct a _light_ theme from the colours I am already generating. Maybe an easy `reverse()` would work. Pretty sure that if I would start reading about colour theory again, I would learn that it's not just that easy.
2. I wonder how I could make use of this module also outside neovim. [Wezterm](https://wezfurlong.org/wezterm/index.html) can execute Lua, or there could be a function that prints out the current generated colour scheme as configuration for a given tool. I like to have the same look in multiple tools I use throughout the day. Any thoughts about this one?

**Comment by echasnovski** [comment]
> 1. Is there a simple way to construct a _light_ theme from the colours I am already generating. Maybe an easy `reverse()` would work. Pretty sure that if I would start reading about colour theory again, I would learn that it's not just that easy.

Use light background and dark foreground. If you want particular hue, I'd recommend using [Oklch interactive picker](https://bottosson.github.io/misc/colorpicker).

For Nord palette it seems to be `{ background = '#ECEFF4', foreground = '#434C5E' }`.

> 2. I wonder how I could make use of this module also outside neovim. [Wezterm](https://wezfurlong.org/wezterm/index.html) can execute Lua, or there could be a function that prints out the current generated colour scheme as configuration for a given tool. I like to have the same look in multiple tools I use throughout the day. Any thoughts about this one?

Yes, there is `make_palette()`. Give it the config as input, and it will return all colors used (and slightly more). Try this, for example: `:= require('mini.hues').make_palette({ background = '#11262d', foreground = '#c0c8cc' })`. If you want it as text in regular buffer, there is `put_text` in 'mini.misc'.

**Comment by Iron-E** [comment]
> Two comments I already have: 

Not the plugin author, but I think I can address these:

1. `mini.colors` can do that (e.g. `chan_invert('lightness', {gamut_clip = 'cusp'})`)
2. I have a [colorscheme exporter](https://github.com/Iron-E/nvim-highlite#exporting-colorschemes) that has Wezterm support ([source code](https://github.com/Iron-E/nvim-highlite/blob/master-v4/lua/highlite/export/wezterm.lua), feel free to repurpose it)

**Comment by echasnovski** [comment]
> 1. `mini.colors` can do that (e.g. `chan_invert('lightness', {gamut_clip = 'cusp'})`)

True, this is also a valid approach. Using manual "light bg, dark fg" colors with `saturation = 'high'` will usually result into cleaner color scheme. Plus you can adjust lightness of both background and text like this.

**Comment by tssm** [comment]
This is awesome! `randomhue` is my main colour scheme now ☺️ I notice though that by default the `StatusLine` highlight looks almost like the `Normal`. Is it intentional? I would like to suggest using the same background as `StatusLineNC`, and change the foreground colour instead to something with higher contrast 🙏

I'm also thinking of submitting a PR with values for `hl-User1..9`, but I would like to know if you agree that changing only the foreground makes sense.

Regardless, this is a great plug-in, and it motivated me to try the other modules in the repo. Thanks so much for it!

**Comment by echasnovski** [comment]
Thanks for kind words!

> I notice though that by default the `StatusLine` highlight looks almost like the `Normal`. Is it intentional? I would like to suggest using the same background as `StatusLineNC`, and change the foreground colour instead to something with higher contrast

Background of `StatusLine` is [accent with background lightness](https://github.com/echasnovski/mini.nvim/blob/afa6f0a3dc27b910501651f5fc26ef3b551fd99b/lua/mini/hues.lua#L705). By default accent uses background hue, so this appears the same as `Normal`.

This is intentional. If you want something different, I'd suggest using custom accent or override `StatusLine` manually (maybe inside `ColorScheme` event).
 
> I'm also thinking of submitting a PR with values for `hl-User1..9`, but I would like to know if you agree that changing only the foreground makes sense.

I am not sure the whole `hl-User1..9` is a good idea, as it seems to mostly be reserved to be defined by the user (as name suggests). I can't remember seeing recent-ish color scheme defining those.

**Comment by tssm** [comment]
> Background of StatusLine is [accent with background lightness](https://github.com/echasnovski/mini.nvim/blob/afa6f0a3dc27b910501651f5fc26ef3b551fd99b/lua/mini/hues.lua#L705). By default accent uses background hue, so this appears the same as Normal

Right, I noticed that. Would it be possible to adjust the `accent` for `randomhue`? I already tried setting it up, but it's ignored. I haven't dived into the code that much, so I'm not sure if what I'm asking would be possible. In any case, I already use `ColorScheme` to adjust some stuff, so I will go that route for now

> I am not sure the whole `hl-User1..9` is a good idea, as it seems to mostly be reserved to be defined by the user (as name suggests). I can't remember seeing recent-ish color scheme defining those.

This may be weird, but I always wanted schemes to define those, so I could just use them 😅 But fair enough, `ColorScheme` is the way to go!

**Comment by echasnovski** [comment]
> Right, I noticed that. Would it be possible to adjust the `accent` for `randomhue`? I already tried setting it up, but it's ignored. I haven't dived into the code that much, so I'm not sure if what I'm asking would be possible. In any case, I already use `ColorScheme` to adjust some stuff, so I will go that route for now

Not for the 'randomhue' per se, but you can create your own randomhue-like color scheme with different background:
- Create file 'colors/my_randomhue.lua' in your home config directory ('~/.config/nvim/' if you are on Unix).
- Copy lines from [original 'randomhue'](https://github.com/echasnovski/mini.nvim/blob/main/colors/randomhue.lua).
- Change `saturation` option to your liking. Also I'd recommend changing `vim.g.colors_name = 'my_randomhue'`.
- Source it with `:colorscheme my_randomhue`.
